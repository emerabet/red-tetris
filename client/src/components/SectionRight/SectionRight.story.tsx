import React from 'react';
import { storiesOf } from '@storybook/react';
import SectionRight from './SectionRight';

storiesOf('SectionRight', module)
  .add('4 other players', () => (
    <SectionRight
    spectres={
      [{
        id: 'oponent 1',
        spectre: '0123456789',
        username: 'aaaaa',
      }, {
        id: 'oponent 2',
        spectre: '0123456789',
        username: 'bbb',
      }]
    }
      pieces={'ZOL'}
      started={true}
    />
  ));
