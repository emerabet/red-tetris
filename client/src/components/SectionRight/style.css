.sectionRight {
	padding-top: 30px;
	padding-bottom: 30px;
	display: flex;
	flex-grow: 1;
	flex-direction: column;
	align-items: flex-start;
	min-width: 250px;
	max-width: 35%;
}

.sectionRightInside {
	min-width: 250px;
	display: flex;
	flex-grow: 1;
	flex-direction: column;
	align-items: center;
}

.opponents {
	display: flex;
	flex-direction: row;
	flex-wrap: wrap;
	align-items: center;
	justify-content: space-around;
}

.opponentsNoWrap {
	display: flex;
	flex-direction: row;
	align-items: center;
	transform: scale(0.7);
}

.lab {
	font-family: 'Raleway Dots', cursive;
	font-size: 30px;
}
