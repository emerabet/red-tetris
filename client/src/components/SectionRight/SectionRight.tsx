import React from 'react';
import RedTetris from '../RedTetris';
import LabeledBox from '../LabeledBox';
import NextPieces from '../NextPieces';
import * as utils from '../../Utils/constants';
import SquareLabel from '../Labels';
import Oponent from '../Oponent';

import { SpectreI } from '../../types/gameTypes';

import './style.css';

interface SectionRightProps {
  spectres: SpectreI[];
  pieces: string;
  started: boolean;
}

const SectionRight: React.SFC<SectionRightProps> = (props) => {

  return (
    <div className="sectionRight">
      {props.started && <div className="sectionRightInside">
        {/* <SquareLabel
          label="next pieces:"
        /> */}
        <div className="lab">NEXT PIECES</div>
        <NextPieces
          pieces={props.pieces}
        />
        {/* <SquareLabel
          label="opponents:"
        /> */}
        <div className="lab">OPPONENTS</div>
        <div className="opponents">
          {
            props.spectres.map((spectre, i) => {
              return <Oponent key={`op_${i}`} spectre={spectre} />;
            })
          }
        </div>
      </div>}
    </div>
  );
};

export default SectionRight;
