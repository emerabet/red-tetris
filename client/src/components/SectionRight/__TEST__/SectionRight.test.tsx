import React from 'react';

import renderer from 'react-test-renderer';
import SectionRight from '../SectionRight';

it('renders correctly with defaults', () => {
  const sectionRight = renderer.create(<SectionRight
    oponents={[{
        name: 'oponenetetete',
        game: '0123456789',
      },
      {
        name: 'booottt',
        game: '4444333345',
      },
      {
        name: 'pop',
        game: '88AAJJIIIJ',
      }, {
        name: 'bobo',
        game: '0040565600',
      }]}
      pieces={'ZOL'}
  />).toJSON();
  expect(sectionRight).toMatchSnapshot();
});
