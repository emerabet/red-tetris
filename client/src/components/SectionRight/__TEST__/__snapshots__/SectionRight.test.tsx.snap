// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<div
  className="sectionRight"
>
  <div
    className="sectionRightInside"
  >
    <div
      className="lab"
    >
      NEXT PIECES
    </div>
    <div
      className="pieceRow"
    >
      <div
        className="nextPiece"
      >
        <div
          className="boardRow"
        >
          <div
            className="boardCell cell1"
          />
          <div
            className="boardCell cell1"
          />
          <div
            className="boardCell cell0"
          />
        </div>
        <div
          className="boardRow"
        >
          <div
            className="boardCell cell0"
          />
          <div
            className="boardCell cell1"
          />
          <div
            className="boardCell cell1"
          />
        </div>
        <div
          className="boardRow"
        >
          <div
            className="boardCell cell0"
          />
          <div
            className="boardCell cell0"
          />
          <div
            className="boardCell cell0"
          />
        </div>
      </div>
      <div
        className="nextPiece"
      >
        <div
          className="boardRow"
        >
          <div
            className="boardCell cell0"
          />
          <div
            className="boardCell cell7"
          />
          <div
            className="boardCell cell7"
          />
        </div>
        <div
          className="boardRow"
        >
          <div
            className="boardCell cell0"
          />
          <div
            className="boardCell cell7"
          />
          <div
            className="boardCell cell7"
          />
        </div>
      </div>
      <div
        className="nextPiece"
      >
        <div
          className="boardRow"
        >
          <div
            className="boardCell cell0"
          />
          <div
            className="boardCell cell0"
          />
          <div
            className="boardCell cell4"
          />
        </div>
        <div
          className="boardRow"
        >
          <div
            className="boardCell cell4"
          />
          <div
            className="boardCell cell4"
          />
          <div
            className="boardCell cell4"
          />
        </div>
        <div
          className="boardRow"
        >
          <div
            className="boardCell cell0"
          />
          <div
            className="boardCell cell0"
          />
          <div
            className="boardCell cell0"
          />
        </div>
      </div>
    </div>
    <div
      className="lab"
    >
      OPPONENTS
    </div>
    <div
      className="opponents"
    >
      <div
        className="opponentFlex"
      >
        <div
          className="opponent"
        >
          <div
            className="opponentColumn opponentColumn_0"
          />
          <div
            className="opponentColumn opponentColumn_1"
          />
          <div
            className="opponentColumn opponentColumn_2"
          />
          <div
            className="opponentColumn opponentColumn_3"
          />
          <div
            className="opponentColumn opponentColumn_4"
          />
          <div
            className="opponentColumn opponentColumn_5"
          />
          <div
            className="opponentColumn opponentColumn_6"
          />
          <div
            className="opponentColumn opponentColumn_7"
          />
          <div
            className="opponentColumn opponentColumn_8"
          />
          <div
            className="opponentColumn opponentColumn_9"
          />
        </div>
        <div
          className="labelTitle small"
        >
          <div>
             
            <div
              className="labelLetterM"
            >
              <div
                className="labelCube whiteCube a1"
              />
              <div
                className="labelCube whiteCube a2"
              />
              <div
                className="labelCube whiteCube a3"
              />
              <div
                className="labelCube whiteCube a4"
              />
              <div
                className="labelCube whiteCube a5"
              />
              <div
                className="labelCube whiteCube a6"
              />
              <div
                className="labelCube whiteCube a7"
              />
              <div
                className="labelCube whiteCube a8"
              />
              <div
                className="labelCube whiteCube a9"
              />
              <div
                className="labelCube whiteCube a10"
              />
              <div
                className="labelCube whiteCube a11"
              />
              <div
                className="labelCube whiteCube a12"
              />
              <div
                className="labelCube whiteCube a13"
              />
            </div>
             
          </div>
          <div>
             
            <div
              className="labelLetterM"
            >
              <div
                className="labelCube whiteCube a1"
              />
              <div
                className="labelCube whiteCube a2"
              />
              <div
                className="labelCube whiteCube a3"
              />
              <div
                className="labelCube whiteCube a4"
              />
              <div
                className="labelCube whiteCube a5"
              />
              <div
                className="labelCube whiteCube a6"
              />
              <div
                className="labelCube whiteCube a7"
              />
              <div
                className="labelCube whiteCube a8"
              />
              <div
                className="labelCube whiteCube a9"
              />
              <div
                className="labelCube whiteCube a10"
              />
              <div
                className="labelCube whiteCube a11"
              />
              <div
                className="labelCube whiteCube a12"
              />
              <div
                className="labelCube whiteCube a13"
              />
            </div>
             
          </div>
          <div>
             
            <div
              className="labelLetterM"
            >
              <div
                className="labelCube whiteCube a1"
              />
              <div
                className="labelCube whiteCube a2"
              />
              <div
                className="labelCube whiteCube a3"
              />
              <div
                className="labelCube whiteCube a4"
              />
              <div
                className="labelCube whiteCube a5"
              />
              <div
                className="labelCube whiteCube a6"
              />
              <div
                className="labelCube whiteCube a7"
              />
              <div
                className="labelCube whiteCube a8"
              />
              <div
                className="labelCube whiteCube a9"
              />
              <div
                className="labelCube whiteCube a10"
              />
              <div
                className="labelCube whiteCube a11"
              />
              <div
                className="labelCube whiteCube a12"
              />
              <div
                className="labelCube whiteCube a13"
              />
            </div>
             
          </div>
          <div>
             
            <div
              className="labelLetterM"
            >
              <div
                className="labelCube whiteCube a1"
              />
              <div
                className="labelCube whiteCube a2"
              />
              <div
                className="labelCube whiteCube a3"
              />
              <div
                className="labelCube whiteCube a4"
              />
              <div
                className="labelCube whiteCube a5"
              />
              <div
                className="labelCube whiteCube a6"
              />
              <div
                className="labelCube whiteCube a7"
              />
              <div
                className="labelCube whiteCube a8"
              />
              <div
                className="labelCube whiteCube a9"
              />
              <div
                className="labelCube whiteCube a10"
              />
              <div
                className="labelCube whiteCube a11"
              />
              <div
                className="labelCube whiteCube a12"
              />
              <div
                className="labelCube whiteCube a13"
              />
            </div>
             
          </div>
          <div>
             
            <div
              className="labelLetterM"
            >
              <div
                className="labelCube whiteCube a1"
              />
              <div
                className="labelCube whiteCube a2"
              />
              <div
                className="labelCube whiteCube a3"
              />
              <div
                className="labelCube whiteCube a4"
              />
              <div
                className="labelCube whiteCube a5"
              />
              <div
                className="labelCube whiteCube a6"
              />
              <div
                className="labelCube whiteCube a7"
              />
              <div
                className="labelCube whiteCube a8"
              />
              <div
                className="labelCube whiteCube a9"
              />
              <div
                className="labelCube whiteCube a10"
              />
              <div
                className="labelCube whiteCube a11"
              />
              <div
                className="labelCube whiteCube a12"
              />
              <div
                className="labelCube whiteCube a13"
              />
            </div>
             
          </div>
        </div>
      </div>
      <div
        className="opponentFlex"
      >
        <div
          className="opponent"
        >
          <div
            className="opponentColumn opponentColumn_0"
          />
          <div
            className="opponentColumn opponentColumn_1"
          />
          <div
            className="opponentColumn opponentColumn_2"
          />
          <div
            className="opponentColumn opponentColumn_3"
          />
          <div
            className="opponentColumn opponentColumn_4"
          />
          <div
            className="opponentColumn opponentColumn_5"
          />
          <div
            className="opponentColumn opponentColumn_6"
          />
          <div
            className="opponentColumn opponentColumn_7"
          />
          <div
            className="opponentColumn opponentColumn_8"
          />
          <div
            className="opponentColumn opponentColumn_9"
          />
        </div>
        <div
          className="labelTitle small"
        >
          <div>
             
            <div
              className="labelLetterM"
            >
              <div
                className="labelCube whiteCube b1"
              />
              <div
                className="labelCube whiteCube b2"
              />
              <div
                className="labelCube whiteCube b3"
              />
              <div
                className="labelCube whiteCube b4"
              />
              <div
                className="labelCube whiteCube b5"
              />
              <div
                className="labelCube whiteCube b6"
              />
              <div
                className="labelCube whiteCube b7"
              />
              <div
                className="labelCube whiteCube b8"
              />
              <div
                className="labelCube whiteCube b9"
              />
              <div
                className="labelCube whiteCube b10"
              />
              <div
                className="labelCube whiteCube b11"
              />
              <div
                className="labelCube whiteCube b12"
              />
              <div
                className="labelCube whiteCube b13"
              />
            </div>
             
          </div>
          <div>
             
            <div
              className="labelLetterM"
            >
              <div
                className="labelCube whiteCube b1"
              />
              <div
                className="labelCube whiteCube b2"
              />
              <div
                className="labelCube whiteCube b3"
              />
              <div
                className="labelCube whiteCube b4"
              />
              <div
                className="labelCube whiteCube b5"
              />
              <div
                className="labelCube whiteCube b6"
              />
              <div
                className="labelCube whiteCube b7"
              />
              <div
                className="labelCube whiteCube b8"
              />
              <div
                className="labelCube whiteCube b9"
              />
              <div
                className="labelCube whiteCube b10"
              />
              <div
                className="labelCube whiteCube b11"
              />
              <div
                className="labelCube whiteCube b12"
              />
              <div
                className="labelCube whiteCube b13"
              />
            </div>
             
          </div>
          <div>
             
            <div
              className="labelLetterM"
            >
              <div
                className="labelCube whiteCube b1"
              />
              <div
                className="labelCube whiteCube b2"
              />
              <div
                className="labelCube whiteCube b3"
              />
              <div
                className="labelCube whiteCube b4"
              />
              <div
                className="labelCube whiteCube b5"
              />
              <div
                className="labelCube whiteCube b6"
              />
              <div
                className="labelCube whiteCube b7"
              />
              <div
                className="labelCube whiteCube b8"
              />
              <div
                className="labelCube whiteCube b9"
              />
              <div
                className="labelCube whiteCube b10"
              />
              <div
                className="labelCube whiteCube b11"
              />
              <div
                className="labelCube whiteCube b12"
              />
              <div
                className="labelCube whiteCube b13"
              />
            </div>
             
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
